ENIGMAsystem/SHELL/Audio_Systems/SDL/SDLbasic.cpp

```diff
@@ -0,0 +1,180 @@
+#include <stdio.h>
+#include <string>
+#include "../General/ASbasic.h"
+#include "Audio_Systems/audio_mandatory.h"
+#include "Widget_Systems/widgets_mandatory.h"
+#include "../General/ASutil.h"
+#include "SoundResource.h"
+#include "SDLsystem.h"
+#include <SDL_mixer.h>
+
+namespace enigma_user {
+
+bool sound_exists(int sound) { return sounds.exists(sound); }
+
+int sound_add(string fname, int kind, bool preload) {
+
+  // Open sound
+  size_t flen = 0;
+  char* fdata = enigma::read_all_bytes(fname, flen);
+  if (!fdata) {
+    DEBUG_MESSAGE(std::string("The Sound file failed - ") + std::string(fname) + std::string(" failed to open"), MESSAGE_TYPE::M_ERROR);
+    return -1;
+  }
+  // Decode sound
+  int rid = sounds.size();
+  bool fail = enigma::sound_add_from_buffer(rid, fdata, flen);
+  delete[] fdata;
+  
+  if (fail) {
+    DEBUG_MESSAGE(std::string("The Sound file failed - ") + std::string(fname) + std::string(" failed to open"), MESSAGE_TYPE::M_ERROR);
+    return -1;
+  }
+  return rid;
+
+}
+
+void sound_delete(int sound) {
+  sound_stop(sound);
+  sounds.destroy(sound);
+}
+
+bool sound_play(int sound) {
+  const Sound& snd = sounds.get(sound);
+  if (Mix_PlayChannel(-1,snd.mc, 0)==-1) { return false; }
+  return true;

```

Comment:
You can actually combine this like `return (Mix_PlayChannel(-1,snd.mc, 0) != -1);` to be simpler and elsewhere.